spam/03_spam_data_slicing_tutorial.py

```diff
@@ -362,80 +249,198 @@ def textblob_polarity(x):
+
+# %%
+S_train = applier.apply(df_train)
+S_valid = applier.apply(df_valid)
+
+valid_dl_slice = slice_model.make_slice_dataloader(valid_dl.dataset, S_valid)
+test_dl_slice = slice_model.make_slice_dataloader(test_dl.dataset, S_test)
+
+# %% [markdown]
+# Then, we can use the [`score_slices`]() in `BinarySliceClassifier` to natively monitor performance by slice.
+#
+# *NOTE:* This method differs from `score` — it remaps all slice-related labels, denoted `spam_task_slice:{slice_name}_pred`, to be evaluated on the `spam_task`.
+
+# %%
+slice_model.score_slices([valid_dl_slice, test_dl_slice], as_dataframe=True)
+
 # %% [markdown]
 # ### Representation learning with slices
 
 # %% [markdown]
 # To cope with scale, we will attempt to learn and combine many slice-specific representations with an attention mechanism.
-# (For details, please see our technical report — coming soon!)
+# (For details about this approach, please see our technical report — coming soon!)
 
 # %% [markdown]
-# First, we update our existing dataloader with our new slices.
+# First, we'll generate the remaining `S` matrixes with the new set of slicing functions.
 
 # %%
 applier = PandasSFApplier(sfs)
 S_train = applier.apply(df_train)
 S_valid = applier.apply(df_valid)
 
+# %% [markdown]
+# We now highlight the slice-aware capabilities of `BinarySlicingClassifier`.
+# At a high-level, this model adds additional capacity corresponding to each slice through additional _slice-specific tasks_.
+
 # %%
-slice_names = [sf.name for sf in sfs]
-add_slice_labels(dl_train, spam_task, S_train, slice_names)
-add_slice_labels(dl_valid, spam_task, S_valid, slice_names)
-add_slice_labels(dl_test, spam_task, S_test, slice_names)
+slice_model = BinarySlicingClassifier(
+    base_architecture=mlp,
+    head_dim=bow_dim,
+    input_data_key="bow_features",
+    task_name="spam_task",
+    slice_names=slice_names,
+)
 
 # %% [markdown]
-# Using the helper, [`convert_to_slice_tasks`](https://snorkel.readthedocs.io/en/redux/packages/_autosummary/slicing/snorkel.slicing.convert_to_slice_tasks.html), we can convert our original `spam_task` into slice-specific tasks.
-# These will be used to learn "expert task heads" for each slice, in the style of multi-task learning.
-# The original `spam_task` now contains the attention mechanism to then combine these slice experts.
+# *Bonus, for the curious:* You'll notice that `BinarySlicingClassifier` is a subclass of [`MultitaskClassifier`](https://snorkel.readthedocs.io/en/redux/packages/_autosummary/classification/snorkel.classification.MultitaskClassifier.html) — in our implementation, the slice model models each slice as slice-specific task-heads:
+# * `ind` heads are trained with slice indicators
+# * `pred` heads are trained on the original task, but _only_ with examples belonging to the slice
 
 # %%
-from snorkel.slicing import convert_to_slice_tasks
+slice_model.task_names
```

Comment:
Skip the above.  At most, briefly mention the connection to MTL model.  A good rule with technical exposition: Either stay high level, or go fully deep enough so the reader / listener can understand... halfway is just confusing with minimal benefit